dotnet/src/Connectors/Connectors.OpenAI.UnitTests/Services/OpenAIChatCompletionServiceTests.cs

```diff
@@ -1179,9 +1179,11 @@ public async Task GetChatMessageContentsSendsValidJsonSchemaForStructuredOutputs
     }
 
     [Theory]
-    [InlineData(typeof(TestStruct))]
-    [InlineData(typeof(TestStruct?))]
-    public async Task GetChatMessageContentsSendsValidJsonSchemaWithStruct(Type responseFormatType)
+    [InlineData(typeof(TestStruct), "TestStruct")]
+    [InlineData(typeof(TestStruct?), "TestStruct")]
+    [InlineData(typeof(TestStruct<string>), "TestStruct1")]
+    [InlineData(typeof(TestStruct<string>?), "TestStruct1")]
```

Comment:
If you add `typeof(TestStruct<float>)` will the expected schema name also be  `"TestStruct1"` and does this matter?

E.g. if `TestStruct<float>` and `TestStruct<string>` are the types of properties on a complex object could this confuse the LLM?

Reply:
> If you add typeof(TestStruct<float>) will the expected schema name also be "TestStruct1" and does this matter?

@markwallace-microsoft Based on my testing, this property is not used by AI model to provide a result.

> E.g. if TestStruct'float and TestStruct'string are the types of properties on a complex object could this confuse the LLM?

This functionality is only about a root type, which is used for schema name. If the types are properties of a root type, these types are not used at all, since all complex objects are defined as `object` in JSON schema.

But in all cases, I do agree that if we have support for this, it would be nice to have a differentiation between `TestStruct<float>` and `TestStruct<string>`, so I updated the logic to include the name of generic arguments as well.